launch/launch/utilities/type_utils.py

```diff
@@ -0,0 +1,565 @@
+    return data_type is List or (
+        hasattr(data_type, '__origin__') and
+        hasattr(data_type, '__args__') and
+        data_type.__origin__ in  # type: ignore
+        (list, List) and  # On Linux/Mac is List, on Windows is list.
+        len(data_type.__args__) > 0 and
+        data_type is List[data_type.__args__[0]]  # type: ignore
+    )
+
+
+def is_valid_scalar_type(data_type: AllowedTypesType) -> bool:
+    """
+    Return `True` if `data_type` is an instance of a valid scalar type.
+
+    See :py:data:`ScalarTypesTuple` for further reference.
+    """
+    return data_type in ScalarTypesTuple
+
+
+def extract_type(data_type: AllowedTypesType) -> Tuple[ScalarTypesType, bool]:
+    """
+    Extract type information from type object.
+
+    :param data_type: It can be:
+        - a scalar type i.e. `str`, `int`, `float`, `bool`;
+        - a uniform list i.e `List[str]`, `List[int]`, `List[float]`, `List[bool]`;
+
+    :returns: a tuple (type_obj, is_list).
+        `is_list` is `True` for the supported list types, if not is `False`.
+        `type_obj` is the object representing that type in Python. In the case of a list,
+        it's the type of the items.
+        e.g.:
+            `data_type = List[int]` -> `(int, True)`
+            `data_type = bool` -> `(bool, False)`
+    """
+    is_list = False
+    scalar_type: ScalarTypesType = cast(ScalarTypesType, data_type)
+    if is_typing_list(data_type):
+        is_list = True
+        scalar_type = data_type.__args__[0]  # type: ignore
+    if is_valid_scalar_type(scalar_type) is False:
+        raise ValueError(f'Unrecognized data type: {data_type}')
+    return (scalar_type, is_list)
+
+
+def is_instance_of_valid_type(value: Any, can_be_str: bool = False) -> bool:
+    """
+    Return `True` if value is an instance of an allowed type.
+
+    See :py:data:`AllowedTypesTuple` definition for further reference.
+
+    :param value: variable to be checked.
+    :param can_be_str: when True, non-uniform lists mixed with strings are allowed.
+    :return: `True` when value is an instance of a valid type, `False` if not.
+    """
+    if isinstance(value, list):
+        if not value:
+            return True  # Accept empty lists.
+        member_types = {type(x) for x in value}
+        if len(member_types) == 2:
+            if can_be_str is False or str not in member_types:
+                return False
+        elif len(member_types) > 2:
+            return False
+        else:
+            assert len(member_types) == 1
+        assert len(member_types) in (1, 2)
+        valid_types = tuple(member_types)
+        member_types.discard(str)
+        try:
+            member_type = member_types.pop()
+        except KeyError:
+            member_type = str
+        return (
+            all(isinstance(x, valid_types) for x in value[1:]) and
+            member_type in ScalarTypesTuple
+        )
+    return isinstance(value, ScalarTypesTuple)
+
+
+def is_instance_of(
```

Comment:
@ivanpauno meta, safe to ignore: this one's name is odd. I'd expect `is_instance_of` to be a variation of `isinstance` that supports a subset of `typing` generics and `is_instance_of_valid_type` to be sugar to call `is_instance_of` with `AllowedTypesTuple`.

Reply:
Hmm, I will ignore it for the moment, but we can improve that in the future.